tests/cupy_tests/math_tests/test_misc.py

```diff
@@ -255,6 +255,24 @@ def test_nan_to_num_broadcast(self, kwarg):
             with pytest.raises(ValueError):
                 xp.nan_to_num(0.0, **{kwarg: y})
 
+    @testing.for_all_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_equal()
+    def test_real_if_close(self, xp, dtype):
+        x = testing.shaped_random((10,), xp, dtype) + 1e-15j
+        return xp.real_if_close(x)
```

Comment:
This test can only check for complex dtypes inputs. For example, integer_array + 1e-15j is complex array.

Reply:
I feel adding another test without any `+ <imag_term>` would work right? Cause that specific test is only required for the line `if not issubclass(a.dtype.type, cupy.complexfloating):`

Reply:
Or should I just tweak this test for the same, as the other 2 tests are present with the `+ <imag_term>` to test the main function of `real_if_close`

Reply:
How about splitting tests for complex dtypes and tests for no-complex dtypes?